connector/datadogconnector/connector.go

```diff
@@ -66,6 +70,7 @@ func newTraceToMetricConnector(set component.TelemetrySettings, cfg component.Co
 		in:              in,
 		metricsConsumer: metricsConsumer,
 		exit:            make(chan struct{}),
+		metricsClient:   datadog.NewMetricClient(set.MeterProvider.Meter("datadogconnector")),
```

Comment:
Same question as with the exporter: do we really need to store the metrics client here? We need to initialize it, but we don't do anything with it explicitly, so we don't have to put it here